Frontend testable code

Kris2ffer opened the pull request:
Functionality has been moved from study page and pod page to it's own file where it is easy to test.

Kris2ffer committed:
Fix for error message

Kris2ffer committed:
Create PodPage.test.js

Kris2ffer committed:
Added code coveredge testing

Kris2ffer committed:
Created service for study

Kris2ffer committed:
Using new study service

Kris2ffer committed:
Added functions for pod rules

Kris2ffer committed:
Using study service

Kris2ffer committed:
Update App.test.js

Kris2ffer committed:
Added comments to dataset functions

Kris2ffer committed:
Update PodPage.test.js

Kris2ffer committed:
Added some tests for study service

Tests for rules, userlist and dataset

Kris2ffer committed:
Added coverege script

Kris2ffer committed:
Delete App.test.js

Kris2ffer committed:
Create App.test.js

Kris2ffer committed:
Removed error logging to console

Kris2ffer committed:
Added test for setting current study

Kris2ffer committed:
Renamed test folder to __tests__

gjersvik commented on FrontEnd/package.json:
```diff
@@ -13,6 +13,7 @@
     "start": "set PORT=8080 && react-scripts start",
     "build": "react-scripts build",
     "test": "react-scripts test",
+    "test:coverege": "react-scripts test --coverage",
```
need to add --watchAll=false for coverage to work.
Also typo in test:coverege.

gjersvik commented on FrontEnd/package.json:
```diff
@@ -29,5 +30,15 @@
+      "!**/coverage/**",
+      "!**/serviceWorker.js",
+      "!**/index.js",
+      "!**/App.js"
```
Have mixed feelings if these files should be ignored here or by ignore comment.
```
/* istanbul ignore file */ to ignore file.
/* istanbul ignore next */ to ignore the next thing. Function, line, class.
```

gjersvik commented on FrontEnd/src/__tests__/App.test.js:
```diff
@@ -0,0 +1,22 @@
+it('renders without crashing', () => {
+  act(() => {
+    ReactDOM.render(<App />, div);
+  })
```
This test results in errors even if it passes that send mixed messages. I would ignore it until you can create is safely.

gjersvik reviewed (approved):
Look good have just added some pointers and thoughts for future work.
One general ting. files that contains react <div ... > syntax should be called .jsx as they are not valid .js files.

Kris2ffer committed:
Update studyService.js

Kris2ffer committed:
Fixed coverege issue

Kris2ffer committed:
Added more coverege and fixed duplication

Kris2ffer committed:
Fixed typo

Kris2ffer committed:
Fixed error when testing app component

The login function throws an error because popups are not implemented.
Added a noLogin prop that can be used when testing.
